pkg/agent/cniserver/server.go

```diff
@@ -535,6 +545,47 @@ func (s *CNIServer) CmdAdd(ctx context.Context, request *cnipb.CniCmdRequest) (*
 	return resultToResponse(cniResult), nil
 }
 
+func (s *CNIServer) updatePodAnnotation(ctx context.Context, status *netdefv1.NetworkStatus, podName, podNamespace string) {
+	const maxRetries = 3
+	const retryInterval = time.Second
+	var pod *v1.Pod
+	var lastErr error
+
+	for retry := 0; retry < maxRetries; retry++ {
+		pod, lastErr = s.kubeClient.CoreV1().Pods(podNamespace).Get(ctx, podName, metav1.GetOptions{})
+		if lastErr == nil || errors.IsNotFound(lastErr) {
+			break
+		}
+
+		if retry < maxRetries-1 {
+			klog.V(2).InfoS("Retrying Pod Get after error", "Pod", klog.KRef(podNamespace, podName), "retry", retry+1, "error", lastErr)
+			time.Sleep(retryInterval)
+		}
+	}
+
+	if lastErr != nil {
+		klog.ErrorS(lastErr, "Failed to get Pod after retries", "Pod", klog.KRef(podNamespace, podName), "retries", maxRetries)
+		return
+	}
+
+	annotations := pod.GetAnnotations()
+	if annotations == nil {
+		return
+	}
+	_, netObjExist := annotations[netdefv1.NetworkAttachmentAnnot]
+	if !netObjExist {
+		klog.V(2).InfoS("Skipping network status update for Pod without secondary-network annotation",
+			"Pod", klog.KObj(pod))
+		return
+	}
```

Comment:
I think you should also add a new boolean in CNIServer to skip update annotation when the SecondaryNetwork feature is not enabled. The check should be added before the line status, err := netdefutils.CreateNetworkStatus(cniResult, cniConfig.Name, true, nil)
I already added the feature enablement boolean here https://github.com/antrea-io/antrea/pull/7144/files#diff-c14b020ddf2ac5e6200f5614ac1070d215ff2a7381325614d8cfcf09c2403d06R126. You can refer to it.

Reply:
updated